packages/azure-kusto-ingest/src/abstractKustoClient.ts

```diff
@@ -45,4 +49,41 @@ export abstract class AbstractKustoClient {
             throw new Error("Client is closed");
         }
     }
+
+    getIngestionEndpoint(clusterUrl?: string): string | undefined {
+        if (!clusterUrl || clusterUrl.includes(INGEST_PREFIX) || this.isReservedHostname(clusterUrl)) {
+            return clusterUrl;
+        }
+        return clusterUrl.replace(PROTOCOL_SUFFIX, PROTOCOL_SUFFIX + INGEST_PREFIX);
+    }
+
+    getQueryEndpoint(clusterUrl?: string): string | undefined {
+        if (clusterUrl && clusterUrl.includes(INGEST_PREFIX)) {
+            return clusterUrl.replace(INGEST_PREFIX, "");
+        }
+        return clusterUrl;
+    }
+
+    isReservedHostname(clusterUrl: string): boolean {
+        const parsedUrl = new URL(clusterUrl);
```

Comment:
What happens if this fails? does it throw? will it throw further down the line?

Reply:
True, i need to add a try catch here. if it throws though (not a valid url), we consider it as not a reserved hostname right? ill just return false

Reply:
sure